packages/compass-query-bar/src/stores/query-bar-reducer.ts

```diff
@@ -187,11 +185,9 @@ type ChangeFieldAction = {
  */
 const emitOnQueryChange = (): QueryBarThunkAction<void> => {
   return (dispatch, getState, { localAppRegistry }) => {
-    const { lastAppliedQuery, fields } = getState();
+    const { fields } = getState();
     const query = pickValuesFromFields(fields);
-    if (lastAppliedQuery === null || isEqual(lastAppliedQuery, query)) {
```

Comment:
@gribnoysup this `lastAppliedQuery` was added in the recent refactor improvements https://github.com/mongodb-js/compass/pull/4138. It seems like things work as expected without it, do you know if there is more context I might be missing for why it was added?

Reply:
This is not new, in the previous implementation it was called `lastExecutedQuery`. Feel free to just remove this code, but it's not the actual root cause of the bug and this causes "query outdated" banner to pop up more often (basically every time you re-run a query which was not a thing until I changed this code)

The actual bug is in a few places here: reset action doesn't reset `lastAppliedQuery` to `null` in state but it should, `isEqual` in this if condition should clearly be `!isEqual`, and `emitOnQueryChange` should be dispatched before apply / reset actions are dispatched (because otherwise we are comparing identical queries)

Reply:
Ah gotcha, thanks. Updated to keep that behavior of not showing the banner too much.

Reply:
I'm kinda amazed that with the amount of mistakes I made here we just spotted this 😆